### PR TITLE
Calculator logic

### PR DESCRIPTION
**Pull request by Timowest12**
In this branch:

- Added the files: logic.js and calculate.js to my files and imported them into my component.
- Made all my buttons interactive using the calculate function.
- added logic to make my calculator smoother.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,42 +1,76 @@
 import React, { Component } from "react";
+import calculate from "../logic/calculate";
+import operate from "../logic/operate";
```
Comment (GzimAsani) [inline src/components/Calculator.js]: - [x] `operate.js` is already being used by `calculate.js`, so there's no need to import it

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
 export class Calculator extends Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {calcoutput: 0,calcobj:{},calcnext:null,calctotal:null};
+  }
+  
   render() {
+    const clickbutton = (inp) => (event) => {     
+      //console.log(calculate({},this.state.calcobj))
```
Comment (GzimAsani) [inline src/components/Calculator.js]: - [x] Please remove the commented-out code.

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,42 +1,76 @@
...
+      let returnobj = calculate(this.state.calcobj,inp);
+      this.state.calcobj = returnobj;
```
Comment (GzimAsani) [inline src/components/Calculator.js]: - [x] You should not mutate state directly. Use setState()  instead

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @Timowest12                             
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Great job with the project :muscle: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (Timowest12) [reply src/components/Calculator.js]: Deleted this.

File: src/components/Calculator.js (hunk 2 above)
Comment (Timowest12) [reply src/components/Calculator.js]: removed all commented code.

File: src/components/Calculator.js (hunk 3 above)
Comment (Timowest12) [reply src/components/Calculator.js]: Changed my code to use setState() in all cases. I noticed mutating it directly updates the state but doesn't rerender the component, I thought this was usefull, will always use setState() from now on. Thanks for pointing that out.

File: src/components/Calculator.js [hunk 4]
```diff
@@ -1,42 +1,75 @@
...
+            <div onClick={clickbutton('3')} className="key key3">3</div>
+            <div onClick={clickbutton('+')} className="key key4 orange">+</div>
           </div>
           <div className="row row5">
-            <div className="key key1 flex2">0</div>
-            <div className="key key2 flex1">1</div>
-            <div className="key key3 flex1 orange">=</div>
+            <div onClick={clickbutton('0')} className="key key1 flex2">0</div>
+            <div onClick={clickbutton('1')} className="key key2 flex1">1</div>
+            <div onClick={clickbutton('=')}  className="key key3 flex1 orange">=</div>
```
Comment (D4nielJ) [inline src/components/Calculator.js]: [OPTIONAL] I'm sorry my partner did not notice this, but there are two `1` keys in the calculator. Please replace one of these by the `.` and make sure that works as expected. 

I will mark this as optional, but I highly encourage you to fix it as it is a requirement that the calculator works as expected.

File: src/components/Calculator.js [hunk 5]
```diff
@@ -1,42 +1,75 @@
...
 
         <div className="keys">
           <div className="row row1">
-            <div className="key key1">ac</div>
-            <div className="key key2">+/-</div>
-            <div className="key key3">%</div>
-            <div className="key key4 orange">/</div>
+            <div onClick={clickbutton('ac')} className="key key1">ac</div>
+            <div onClick={clickbutton('+/-')} className="key key2">+/-</div>
+            <div onClick={clickbutton('%')} className="key key4">%</div>
```
Comment (D4nielJ) [inline src/components/Calculator.js:48]: [OPTIONAL] There is a bug when one tries to use the modulo operator with 0, as it is a division by 0. The original logic files have a catch for the division by 0, but not for the modulo. I think you could make it work just like that one.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @Timowest12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 4 above)
Comment (Timowest12) [reply src/components/Calculator.js]: fixed it, can't believe i missed that

File: src/components/Calculator.js (hunk 5 above)
Comment (Timowest12) [reply src/components/Calculator.js:48]: I fixed the bug, but i fixed it inside the operate.js file. I'm not sure if i should edit that file. If not, please let me know.

**Pull request merged**